src/qibolab/platforms/abstract.py

```diff
@@ -15,6 +18,68 @@
 )
 
 
+@dataclass
+class Qubit:
+    name: str
+    readout_frequency: int = 0
+    drive_frequency: int = 0
+    sweetspot: float = 0
+    peak_voltage: float = 0
+    pi_pulse_amplitude: float = 0
+    T1: int = 0
+    T2: int = 0
+    state0_voltage: int = 0
+    state1_voltage: int = 0
+    mean_gnd_states: complex = 0 + 0.0j
+    mean_exc_states: complex = 0 + 0.0j
+    resonator_polycoef_flux: List[float] = field(default_factory=list)
+
+
+@dataclass
+class ExecutionResults:
+    """Data strucutre to store output from `execute_pulse_sequence`."""
```

Comment:
This looks almost exactly the same with what I had in mind :ok_hand: I will confirm without almost when I merge to my QM branch.

Reply:
Since this is a very simple object, I proposed @andrea-pasquale to turn this into a NumPy [structured type](https://numpy.org/doc/stable/user/basics.rec.html#structured-datatypes) (a [`np.dtype`](https://numpy.org/doc/stable/reference/generated/numpy.dtype.html)), such that you will be able to make NumPy arrays of `ExecutionResults`.

In particular:
```py
ExecRes = np.dtype([('i', np.float64), ('q', np.float64), ('sample', np.uint32)])
```

Reply:
Then, if you want methods to operate on these arrays you can make a wrapper class:
```py
@dataclass
class ExecContainer:
    array: npt.NDArray[ExecRes]
```

But you don't have to: you can just have the functions you need in a module. And luckily, in Python modules already make namespace, so you do not need a class to create a scope.

Reply:
> Since this is a very simple object, I proposed @andrea-pasquale to turn this into a NumPy [structured type](https://numpy.org/doc/stable/user/basics.rec.html#structured-datatypes) (a [`np.dtype`](https://numpy.org/doc/stable/reference/generated/numpy.dtype.html)), such that you will be able to make NumPy arrays of `ExecutionResults`.
> 
> In particular:
> 
> ```python
> ExecRes = np.dtype([('i', np.float64), ('q', np.float64), ('sample', np.uint32)])
> ```

I was going to comment about this. For the time being I was using the `dataclass` approach since I was not sure about how to implement the additional functions. Now it is more clear. 


> Then, if you want methods to operate on these arrays you can make a wrapper class:
> 
> ```python
> @dataclass
> class ExecContainer:
>     array: npt.NDArray[ExecRes]
> ```
> 
> But you don't have to: you can just have the functions you need in a module. And luckily, in Python modules already define namespaces, so you do not need a class to create a scope.

With this approach it will be possible to have a symmetry between the original values (`i`, `q` and `samples`) and the ones that we will compute after such as `msr`. As @stavros11 was saying I would like to avoid to call a specific function to obtain those values.  I would like to to results.msr or results.i. What do you think?

Reply:
> I would like to to results.msr or results.i. What do you think?

As I also said in the [comment below](https://github.com/qiboteam/qibolab/pull/262#discussion_r1083762256), I would *consider* the option of dropping the class. But it is perfectly fine deciding to keep it as a container :)

Reply:
(class names above are just placeholders, I didn't think more than 2s for each of them - if you can improve them)